### DISCUSSION TITLE
Bypass cache on loading tabs

### DISCUSSION THREAD

**Discussion by N3C2L** [madprops/grasshopper#158] [Ideas]
I have the following problem, but it's also a possible idea/solution inside 👍 

When I load unloaded tabs, the cached version of the website is displayed, and the scroll position is unfortunately also in the same place (e.g. at the very bottom of the website = where I left the website and unloaded the tab afterwards). 

You can observe this by doing a simple google search with the term "actual time" (google will display the actual time in the search results on top), then unload the tab and revisit the tab some minutes later = it displays not the refreshed time on top of the google search result.

This is the native behavior of Firefox when loading unloaded tabs. But I find this a bit too limited: For example, if I browse a news site, then unload this news site with this tab. A day later, the news site is displayed again, but the scroll position is at the bottom, and the page is loaded from the cache. 

The good thing is, when I refresh the page with Ctrl+F5, both the cache and the scroll position are reset, so the scroll position is at the top again on the website. There is a Firefox addon called "Hard Refresh Button", which brings the Ctrl+F5 function to a button:  

[https://github.com/electrotype/firefox-hard_refresh_button](https://github.com/electrotype/firefox-hard_refresh_button) 

The problem is, after each reload of a previously unloaded tab, I have to press either this button from the addon or Ctrl+F5 afterwards to make the non-cached version visible. 

That's why I wanted to politely ask if you might want to offer an option for GH: When loading a tab (which was previously unloaded) e.g. by mouse clicking on an unloaded tab, the function from above (Ctrl+F5, also called "Hard Refresh") is automatically applied.

I tried it with a combo: Item action + Reload page. The problem with this combos is, that a click on a loaded tab would also reload this loaded tab, even though I just wanted to navigate to this loaded tab without reloading them. And the "Reload Page" command is the same as if I pressed F5, so it loads from the cache. This also the standard behavior of Firefox, that when loading the unloaded tab it's the same as someone would press the F5 key on click. But as written above, Ctrl+F5 would be much better for me. 

Therefore, it would be very kind if you could take a look at this.

**Comment by madprops** [comment]
I'm adding a `Hard Reload` command.

If you want to make it into a combo, you could use commands like `Select Item` -> `Hard Reload`

Or `Soft Action` -> `Hard Reload`

To select item and then hard reload.

**Comment by N3C2L** [comment]
Thank you! The Hard Reload cmd is great, I will use it for sure! 👍 

If I may add one more thing:
The "Item Action" cmd recognizes whether it's a loaded or unloaded tab when clicking on a tab, right?

Because I still have the following problem:
Currently, I have the "Item Action" associated here cmd under Settings > Triggers > Click Items (Tabs). That's the default cmd, which is great.

But what if I always want to perform a "Hard Reload" with a normal left-click on an unloaded tab? This might be possible with a new settings checkbox that includes this function:

"Always do a hard reload on item action on unloaded tabs". With this activated, loaded tabs won't get the hard reload on left click, but unloaded tabs will get the hard reload on left click.

So a user can then decide if they want a hard reload on item action or not with this checkbox.

That would be great, if you could add this. 👍

**Comment by madprops** [comment]
I think it would be better to create a `Click Unloaded Tab` command setting so you can set it to whatever you want. I'll add it in Triggers. I'll make it none by default so it uses the default action, if defined it overrides it. I'll add middle click and others as well for moar settings

I'm also adding a `Hard Action` command that selects a tab and hard reloads it.

**Reply by N3C2L** [reply]
That's awesome! Thanks!

**Comment by N3C2L** [comment]
This idea has been implemented and I guess I can close this.
